File: book_index_manager/exceptions.py
class BookIndexError(Exception):
    """Base exception for all book-index-manager errors."""
    pass

class StorageError(BookIndexError):
    """Raised when storage operations fail."""
    pass

class IdGenerationError(BookIndexError):
    """Raised when ID generation fails."""
    pass

class ConfigError(BookIndexError):
    """Raised when configuration is invalid."""
    pass

class MigrationError(BookIndexError):
    """Raised when data migration fails."""
    pass

File: ui/src/core/exceptions.ts
/**
 * Exception classes for book-index-manager.
 * Port of Python book_index_manager.exceptions
 */

export class BookIndexError extends Error {
    constructor(message?: string) {
        super(message);
        this.name = 'BookIndexError';
    }
}

export class StorageError extends BookIndexError {
    constructor(message?: string) {
        super(message);
        this.name = 'StorageError';
    }
}

export class IdGenerationError extends BookIndexError {
    constructor(message?: string) {
        super(message);
        this.name = 'IdGenerationError';
    }
}

export class ConfigError extends BookIndexError {
    constructor(message?: string) {
        super(message);
        this.name = 'ConfigError';
    }
}

export class MigrationError extends BookIndexError {
    constructor(message?: string) {
        super(message);
        this.name = 'MigrationError';
    }
}
